src/Ng/ElfieFromCatalogCollector.cs

```diff
@@ -221,5 +230,89 @@ public ElfieFromCatalogCollector(Uri index, Storage storage, int maxThreads, Fun
 
             return packageResourceUri;
         }
+
+        /// <summary>
+        /// Decompresses a nupkg file to a temp directory, runs elfie to create an Idx file for the package, and stores the Idx file.
+        /// </summary>
+        async Task<Uri> DecompressAndIndexPackageAsync(Uri packageResourceUri, CatalogItem catalogItem, CancellationToken cancellationToken)
+        {
+            Trace.TraceInformation("#StartActivity DecompressAndIndexPackageAsync " + catalogItem.PackageId + " " + catalogItem.PackageVersion);
+
+            Uri idxResourceUri = null;
+
+            // This is the pointer to the package file in storage
+            Trace.TraceInformation("Loading package from storage.");
+            StorageContent packageStorage = this._storage.Load(packageResourceUri, new CancellationToken()).Result;
```

Comment:
We do store the packages in blob storage. I was intending to store the package, idx, ardb(txt) and patches in blob storage.
It's not a requirement that we store the packages though. I was just keeping them for debugging purposes (if the idx creation fails, we'd have the exact package we were trying to index for investigation.) We can not store them if we end up never using them.